fms/utils/serialization.py

```diff
@@ -88,12 +79,21 @@ def register_adapter(
 
     # Create a new base adapter for this source
     step_functions = [__adapter_steps[architecture][step] for step in adapter_steps]
-    adapter = lambda initial_sd, extra_args: reduce(
-        lambda state_dict, step_func: step_func(state_dict, extra_args),
-        step_functions,
-        initial_sd,
-    )
-    sources[source] = adapter
+
+    def adapter_fn(initial_sd: Mapping[str, Any], **extra_kwargs) -> Mapping[str, Any]:
+        def reduce_fn(
+            state_dict: Mapping[str, Any],
+            step_func: Callable[[Mapping[str, Any]], Mapping[str, Any]],
+        ) -> Mapping[str, Any]:
+            return step_func(state_dict, **extra_kwargs)
```

Comment:
Maybe this could be as a follow-up PR, but given adapting only happens in the beginning on model load, we may want to inspect the adapter step function and only pass in those params it takes, this way we don't need to always create an adapter function with **kwargs